numba/tests/test_parallel_backend.py

```diff
@@ -499,8 +499,42 @@ def generate(cls):
             cls._inject(backend, backend_guard)
 
 
+TestThreadingLayerSelection.generate()
+
+
+@parfors_skip_unsupported
+@skip_unless_py3
+class TestMiscBackendIssues(ThreadLayerTestHelper):
+    """
+    Checks fixes for the issues with threading backends implementation
+    """
+    _DEBUG = False
+
+    @skip_no_omp
+    def test_omp_stack_overflow(self):
+        """
+        Tests that OMP does not overflow stack
+        """
+        runme = """if 1:
+            from numba import vectorize, threading_layer
+            import numpy as np
+            @vectorize(['f4(f4,f4,f4,f4,f4,f4,f4,f4)'], target='parallel')
+            def foo(a,b,c,d,e,f,g,h): 
+                return a+b+c+d+e+f+g+h
+            x = np.ones(2**20, np.float32)
+            foo(*([x]*8))
+            print(threading_layer())
+        """
+        cmdline = [sys.executable, '-c', runme]
+        env = os.environ.copy()
+        env['NUMBA_THREADING_LAYER'] = "omp"
+        env['OMP_STACKSIZE'] = "100K"
+        out, err = self.run_cmd(cmdline, env=env)
+        if self._DEBUG:
+            print(out, err)
+        assert out.strip() == "omp"
```

Comment:
Perhaps check `self.assertIn("omp", out.strip())` as `STDOUT` may not be consistent?